pallets/amm/src/lib.rs

```diff
@@ -74,6 +80,16 @@ pub mod pallet {
         /// A configuration flag to enable or disable the creation of new pools by "normal" users.
         #[pallet::constant]
         type AllowPermissionlessPoolCreation: Get<bool>;
+
+        /// Defines the fees taken out of each trade and sent back to the AMM pool,
+        /// typically 0.3%.
+        type LpFee: Get<Perbill>;
+
+        /// How much the protocol is taking out of each trade.
+        type ProtocolFee: Get<Perbill>;
+
+        // -- TODO --
+        // type ProtocolFeeHandler: OnUnbalanced<CreditOf<Self, I>>;
```

Comment:
Both ORML Tokens and Parity's Assets support the new `fungibles::*` traits from Substrate. We can use those to generate the `CreditOf` Imbalance and pass it to an `OnUnbalanced` runtime implementation

Reply:
@ETeissonniere can you help me determine what type `ProtocolFeeHandler` should be? I've tried the following: 
```rust
type ProtocolFeeHandler: OnUnbalanced<CreditOf<Self::AccountId, Self>>;
```

but get this error:
```rust
error[E0277]: the trait bound `Self: hidden_include::traits::fungibles::Unbalanced<<Self as frame_system::Config>::AccountId>` is not satisfied
```
Thank you!

Reply:
@drbh I think the last `Self` should actually be the tokens pallet